### PR TITLE
container parallel testing

### PR DESCRIPTION
**Pull request by ansvu**
build-depends: 31590
build-depends: https://github.com/dci-labs/bos2-ci-config/pull/148
TestBos2Sno: sno control-plane-1-parallel

### CODE REVIEW

**Review by tkrishtop — COMMENTED** [review]
Hi @ansvu could you please run some test jobs to compare the "before" (DCI main branch) and "after" (current PR) job duration?

**Comment by tkrishtop** [comment]
Hi @ansvu could you please rebase the change to pick the fix for the DCI job failure you're facing?

**Comment by ansvu** [comment]
> Hi @ansvu could you please rebase the change to pick the fix for the DCI job failure you're facing?

Just got  back from PTO. One question, I did not do PR on gerrit to update tests.yml, this PR is failed to at generic test case and it is not even used my changes yet, right?  It failed because I did not do PR on gerrit then using test-hints to test with depend-on PR309? Thanks.

**Comment by manurodriguez** [comment]
recheck

File: roles/preflight/defaults/main.yml [hunk 1]
```diff
@@ -8,4 +8,6 @@ preflight_podman_ca: ""
 catalog_url: "https://catalog.redhat.com/api/containers/v1"
 preflight_test_certified_image: false
 preflight_run_health_check: true
+do_container_parallel_test: false
```
Comment (ramperher) [inline roles/preflight/defaults/main.yml:11]: I don't think this is the best place to have this variable defined. This should be on `dci-openshift-app-agent`, as commented in the change you have opened on Gerrit.

**Comment by tkrishtop** [comment]
Hi @ansvu, I reviewed the PRs and created two more pipelines for the testing.
Here is the complete list of what would be nice to do:

- Close this PR and open another one, creating branch `redhatci:container-parallel` instead of `ansvu:container-parallel`. That would allow me and @manurodriguez modify the commits in the PR and thus speed-up the work.
- Rebase the branch to fix local/check rhel8-build
- Remove build-depends description from [CR 31590](https://softwarefactory-project.io/r/c/dci-openshift-app-agent/+/31590) to avoid circular dependencies
- Run three tests:

1. current preflight tests on your images but with no parallelization, your code is not used: ```build-depends: 31590
build-depends: https://github.com/dci-labs/bos2-ci-config/pull/148
TestBos2Sno: sno control-plane-no-parallel
```

2. parallelization by batches of 1 image: ```build-depends: 31590
build-depends: https://github.com/dci-labs/bos2-ci-config/pull/148
TestBos2Sno: sno control-plane-1-parallel
```

3. parallelization by batches of 3 images: ```build-depends: 31590
build-depends: https://github.com/dci-labs/bos2-ci-config/pull/148
TestBos2Sno: sno control-plane-3-parallel
```

**Comment by manurodriguez** [comment]
retest

**Comment by ansvu** [comment]
retest

**Comment by ansvu** [comment]
retest

**Comment by ansvu** [comment]
retest

**Comment by ansvu** [comment]
Moved to the redhatci:container-parallel

**Pull request closed without merging**